### DISCUSSION TITLE
Community Theme Vote

### DISCUSSION THREAD

**Discussion by jesterb0206** [anuraghazra/github-readme-stats#2268] [Ideas]
I was wondering what it actually takes for a theme to get merged into main? There's pull requests for themes from a year ago or longer that are just sitting there waiting to be merged or shot down. I was thinking the administrator(s) could start a community vote where all pull requests for themes that pass the GitHub Action tests could be voted on and the top 5 or 10 could be merged into main (only a small handful could get approved to avoid bloating). You could give the community say 1 month to vote on themes and then the ones that weren't in the top 5 or 10 could get removed (would help to get rid of a lot of open pull requests). That was just an idea I had, personally I think the pull requests are cluttered and messy right now. You could also have a section in the repository's main README where you show what each of the themes that are up for vote look like and allow each GitHub user that votes to cast votes for 5-10 of their favorite themes. @rickstaa @anuraghazra

**Comment by rickstaa** [comment]
Hey @jesterb0206. Thanks for your suggestion everything that helps to clean up the PRs is an improvement. 🔥 

I implemented the [top-issues-dashboard](https://github.com/anuraghazra/github-readme-stats/blob/master/.github/workflows/top-issues-dashboard.yml), [stale-theme-pr-closer.yaml](https://github.com/anuraghazra/github-readme-stats/blob/master/.github/workflows/stale-theme-pr-closer.yaml) and improved the [preview-theme.yml](https://github.com/anuraghazra/github-readme-stats/blob/master/.github/workflows/preview-theme.yml) to do just that some time ago. These changes allowed me to trim the open themes PRs down from more than 140 to about 60. 

We currently have no written procedure for merging themes into the main branch. 😅 I currently check the themes once a month and merge the ones that receive the most likes, adhere to the theme guidelines and are not similar to the themes we already have. The last ones that were merged were:
- https://github.com/anuraghazra/github-readme-stats/pull/892
- https://github.com/anuraghazra/github-readme-stats/pull/1010

The ones that are most close to being merged are:
- https://github.com/anuraghazra/github-readme-stats/pull/1264
- https://github.com/anuraghazra/github-readme-stats/pull/2264

If these receive more than 3 likes, I will take a look. Maybe we can extend the https://github.com/rickstaa/top-issues-action action such that users can specify custom labels for creating sections. 🤔 Doing this would allow us to show the top-voted themes (i.e. `themes` label), increasing the chance that people will vote for these themes and collaborators will merge them into the main branch. I'm not a fan of manually creating a theme poll every month, but if you can think of another automated solution, don't hesitate to share 👍🏻.

**Reply by rickstaa** [reply]
By the way, I just merged your theme into the main branch since it met all the criteria 🚀. Thanks for your contribution.

**Reply by rickstaa** [reply]
Additionally, I think switching to an organization might make the repository more manageable (see https://github.com/anuraghazra/github-readme-stats/issues/1867). Such a structure would allow us to include more maintainers since organizations allow for more precise permission management.

**Reply by jesterb0206** [reply]
Thank you! I hope I didn't sound too grouchy in my message, I think I woke up on the wrong side of the bed today 😅 But thank you again for all that you and @anuraghazra do in maintaining and improving this repo. I know it's a huge undertaking and as a user I love being able to display my GitHub README stats on my profile README, it inspires me to code on days when I'm exhausted or not in the mood because I get a little ego kick out of seeing my stats go up (I hope that's not bad to admit 😂)! 👍🏼

**Reply by jesterb0206** [reply]
And I like your idea of users being able to specify custom labels for creating sections, manually having to create a poll every month is a lot of work on top of what is already required to maintain GitHub README Stats! 😎

**Reply by jesterb0206** [reply]
@rickstaa While I have you here, I have an off topic question I think you might know the answer too. I forked this repository and I manually update my branch from time to time to keep it up to date with anuraghazra:master. Is there a way I can automate this so I don't have to manually update my branch, I was hoping there was a way you could sync your fork automatically whenever a change was made on the master branch.

**Reply by rickstaa** [reply]
> Thank you! I hope I didn't sound too grouchy in my message, I think I woke up on the wrong side of the bed today sweat_smile But thank you again for all that you and @anuraghazra do in maintaining and improving this repo. I know it's a huge undertaking and as a user I love being able to display my GitHub README stats on my profile README, it inspires me to code on days when I'm exhausted or not in the mood because I get a little ego kick out of seeing my stats go up (I hope that's not bad to admit joy)! 👍🏼

Ha, don't worry, you didn't sound like that. Maybe my message was a bit formal since it was late at night 😅. Great to hear that you enjoy GRS. 🎉  And thanks for being a contributor 🚀!

**Reply by rickstaa** [reply]
> And I like your idea of users being able to specify custom labels for creating sections, manually having to create a poll every month is a lot of work on top of what is already required to maintain GitHub README Stats! sunglasses

I will create a feature request at https://github.com/rickstaa/top-issues-action. Maybe we can also take your suggestion and close theme PRs older than 1 year that did not receive any 👍🏻. People can then decide whether they want to reopen these PRs.

**Reply by rickstaa** [reply]
> @rickstaa While I have you here, I have an off topic question I think you might know the answer too. I forked this repository and I manually update my branch from time to time to keep it up to date with anuraghazra:master. Is there a way I can automate this so I don't have to manually update my branch, I was hoping there was a way you could sync your fork automatically whenever a change was made on the master branch.

This was a often-asked question so I recently added it to the documentation (see https://github.com/anuraghazra/github-readme-stats#keep-your-fork-up-to-date). Feel free to let me know if the solution provided there works.

**Reply by rickstaa** [reply]
@jesterb0206 I created a feature request that would allow us to add a Top Themes section (see https://github.com/rickstaa/top-issues-action/issues/110). I, however, am a bit limited in time at the moment, so I can not give any promises when I implement this feature. 😅 I added the `help-wanted` label so other people can work on it.

**Reply by rickstaa** [reply]
I also created https://github.com/anuraghazra/github-readme-stats/issues/2270 to track the improvements you suggested. If you have the time and would like to contribute more, you can take on some of the TODOs. 👍🏻

**Reply by jesterb0206** [reply]
@rickstaa No worries! I'm busy myself right now too so I understand. I appreciate you creating the feature request, hopefully I'll have more time to contribute soon!

**Reply by rickstaa** [reply]
@jesterb0206 just a heads-up that team voting will be implemented when #2711 is merged.
